src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
```

Comment:
4) As for _map_ in comment 1, it should **not** be static.

5) There is nothing related to the amount of discount: is it always the same, or are we missing some information here?